libs/pages/settings/src/lib/feature/page-organization-helm-repositories-feature/crud-modal-feature/crud-modal-feature.tsx

```diff
@@ -0,0 +1,79 @@
+import { type HelmRepositoryRequest, type HelmRepositoryResponse } from 'qovery-typescript-axios'
+import { useEffect, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import {
+  useAvailableHelmRepositories,
+  useCreateHelmRepository,
+  useEditHelmRepository,
+} from '@qovery/domains/organizations/feature'
+import { CrudModal } from '../../../ui/page-organization-helm-repositories/crud-modal/crud-modal'
+
+export interface CrudModalFeatureProps {
+  onClose: () => void
+  organizationId?: string
+  repository?: HelmRepositoryResponse
+}
+
+export function CrudModalFeature(props: CrudModalFeatureProps) {
+  const { organizationId = '', onClose, repository } = props
+
+  const { data: availableHelmRepositories = [] } = useAvailableHelmRepositories()
+  const { mutateAsync: editHelmRepository } = useEditHelmRepository()
+  const { mutateAsync: createHelmRepository } = useCreateHelmRepository()
+  const [loading, setLoading] = useState(false)
+
+  const methods = useForm({
+    defaultValues: {
+      name: repository?.name,
+      description: repository?.description,
+      url: repository?.url,
+      kind: repository?.kind,
+      config: {},
+    },
+    mode: 'onChange',
+  })
+
+  useEffect(() => {
+    setTimeout(() => methods.clearErrors('config'), 0)
+  }, [methods])
+
+  const onSubmit = methods.handleSubmit(async (data) => {
+    setLoading(true)
+
+    try {
+      if (repository) {
+        await editHelmRepository({
+          organizationId: organizationId,
+          helmRepositoryId: repository.id,
+          helmRepositoryRequest: data as HelmRepositoryRequest,
+        })
+        onClose()
+      } else {
+        await createHelmRepository({
+          organizationId: organizationId,
+          helmRepositoryRequest: data as HelmRepositoryRequest,
+        })
+        onClose()
```

Comment:
You can have only one `onClose()` outside your if

Reply:
https://github.com/Qovery/console/pull/993/commits/6aebbac4f66c4f20666a627ef5e7583f0ae97c2a